pkg/fileutils/fileutils_test.go

```diff
@@ -335,7 +327,7 @@ func TestMatches(t *testing.T) {
 		{"dir/**", "dir/dir2/file", true},
 		{"dir/**", "dir/dir2/file/", true},
 		{"**/dir2/*", "dir/dir2/file", true},
-		{"**/dir2/*", "dir/dir2/file/", false},
+		{"**/dir2/*", "dir/dir2/file/", true},
```

Comment:
These test changes appear to be because the previous version checked internal function that behaved differently than the public ones. Public method also checked the parents of the checked path. https://github.com/docker/docker/blob/3fe2730ab3d7bba483d7bf4fb1e95b627823ff4d/pkg/fileutils/fileutils.go#L106